client/extensions/woocommerce/woocommerce-services/views/label-settings/index.js

```diff
@@ -1,109 +1,110 @@
 
-		const renderCard = ( card, index ) => {
-			const onSelect = () => {
-				this.selectCard( index );
-			};
+		const renderContent = () => {
+			if ( ! formData && ! formMeta.isFetching ) {
+				return (
+					<p>
+						{ translate( 'Unable to get your settings. Please refresh the page to try again.' ) }
+					</p>
+				);
+			}
 
-			return ( <ShippingCard
-				key={ index }
-				onSelect={ onSelect }
-				{ ...card } /> );
+			return (
+				<LabelSettings
+					isLoading={ formMeta.isFetching }
+					pristine={ formMeta.pristine }
+					paymentMethods={ formMeta.payment_methods || [] }
+					setFormDataValue={ setFormDataValue }
+					selectedPaymentMethod={ ( formData || {} ).selected_payment_method_id }
+					paperSize={ ( formData || {} ).paper_size }
+					storeOptions={ storeOptions }
+				/>
+			);
 		};
 
+		const renderToggle = formData && isBoolean( formData.enabled );
+
 		return (
 			<div>
 				<ExtendedHeader
 					label={ translate( 'Shipping Labels' ) }
 					description={ translate( 'Print shipping labels yourself and save a trip to the post office.' ) }>
-					<FormToggle onChange={ onToggle } checked={ this.state.visible } />
+					{ renderToggle && <FormToggle checked={ formData.enabled } onChange={ onEnabledToggle } /> }
 				</ExtendedHeader>
-				<Card className={ classNames( 'shipping__labels-container', { hidden: ! this.state.visible } ) }>
-					<FormFieldSet>
-						<FormLabel
-							className="label-settings__labels-paper-size"
-							htmlFor="paper-size">
-							{ translate( 'Paper size' ) }
-						</FormLabel>
-						<FormSelect name="paper-size">
-							<option>{ translate( 'Letter' ) }</option>
-							<option>{ translate( 'Legal' ) }</option>
-							<option>{ translate( 'Label (4"x6")' ) }</option>
-							<option>{ translate( 'A4' ) }</option>
-						</FormSelect>
-					</FormFieldSet>
-					<FormFieldSet>
-						<FormLabel
-							className="label-settings__cards-label">
-							{ translate( 'Credit card' ) }
-						</FormLabel>
-						<p className="label-settings__credit-card-description">
-							{ translate( 'Use your credit card on file to pay for the labels you print or add a new one.' ) }
-						</p>
-						{ this.state.cards.map( renderCard ) }
-						<Button compact>{ translate( 'Add another credit card' ) }</Button>
-					</FormFieldSet>
+				<Card className={ classNames( 'label-settings__labels-container', { hidden: false } ) }>
+					{ renderContent() }
 				</Card>
 			</div>
 		);
 	}
 }
 
-export default localize( LabelSettings );
+AccountSettingsRootView.propTypes = {
+	submit: PropTypes.func,
+};
+
+function mapStateToProps( state ) {
+	return {
+		siteId: getSelectedSiteId( state ),
+		storeOptions: getLabelSettingsStoreOptions( state ),
+		formData: getLabelSettingsFormData( state ),
+		formMeta: getLabelSettingsFormMeta( state ),
+	};
+}
+
+function mapDispatchToProps( dispatch ) {
+	return {
+		actions: bindActionCreators( settingsActions, dispatch ),
```

Comment:
Our pattern (and Calypso's in general) has been to pass actions in directly as props, rather than under `this.props.actions`.